packages/flutter/lib/src/cupertino/scrollbar.dart

```diff
@@ -91,17 +146,85 @@ class _CupertinoScrollbarState extends State<CupertinoScrollbar> with TickerProv
     return ScrollbarPainter(
       color: _kScrollbarColor,
       textDirection: _textDirection,
-      thickness: _kScrollbarThickness,
+      thickness: _thickness,
       fadeoutOpacityAnimation: _fadeoutOpacityAnimation,
       mainAxisMargin: _kScrollbarMainAxisMargin,
       crossAxisMargin: _kScrollbarCrossAxisMargin,
-      radius: _kScrollbarRadius,
+      radius: _radius,
       padding: MediaQuery.of(context).padding,
       minLength: _kScrollbarMinLength,
       minOverscrollLength: _kScrollbarMinOverscrollLength,
     );
   }
 
+  // Handle a gesture that drags the scrollbar by the given amount.
+  void _dragScrollbar(double primaryDelta) {
+    assert(widget.controller != null);
+
+    // Convert primaryDelta, the amount that the scrollbar moved, into the
+    // coordinate space of the scroll position, and scroll to that
+    // position.
+    final double scrollOffsetLocal = _painter.getTrackToScroll(primaryDelta);
+    _dragScrollbarStartY ??= widget.controller.position.pixels;
+    widget.controller.position.jumpTo(
+      _dragScrollbarStartY + scrollOffsetLocal,
+    );
+  }
+
+  void _startFadeoutTimer() {
+    _fadeoutTimer?.cancel();
+    _fadeoutTimer = Timer(_kScrollbarTimeToFade, () {
+      _fadeoutAnimationController.reverse();
+      _fadeoutTimer = null;
+    });
+  }
+
+  // Long press event callbacks handle the gesture where the user long presses
+  // on the scrollbar thumb and then drags the scrollbar without releasing.
+  void _handleLongPressStart(LongPressStartDetails details) {
+    _fadeoutTimer?.cancel();
+    _fadeoutAnimationController.forward();
+  }
+
+  void _handleLongPress() {
+    _fadeoutTimer?.cancel();
+    _thicknessAnimationController.forward();
+  }
+
+  void _handleLongPressMoveUpdate(LongPressMoveUpdateDetails details) {
+    _dragScrollbar(details.localOffsetFromOrigin.dy);
+  }
+
+  void _handleLongPressEnd(LongPressEndDetails details) {
+    _startFadeoutTimer();
+    _thicknessAnimationController.reverse();
+    _dragScrollbarStartY = null;
+    final ScrollPositionWithSingleContext scrollPosition = widget.controller.position;
+    scrollPosition.goBallistic(details.velocity.pixelsPerSecond.dy);
+  }
+
+  // Horizontal drag event callbacks handle the gesture where the user swipes in
+  // from the right on top of the scrollbar thumb and then drags the scrollbar
+  // without releasing.
+  void _handleHorizontalDragStart(DragStartDetails details) {
+    _horizontalDragScrollbarStartY = details.localPosition.dy;
+    _fadeoutTimer?.cancel();
+    _thicknessAnimationController.forward();
+  }
+
+  void _handleHorizontalDragUpdate(DragUpdateDetails details) {
+    _dragScrollbar(details.localPosition.dy - _horizontalDragScrollbarStartY);
```

Comment:
If the scrollbar scrolls horizontally then I think this doesn't make sense?

Reply:
I've had horizontal scrollbar dragging in the back of my mind while working on this PR...  I just tried to find an instance of it in iOS13 and couldn't.  I'm going to explicitly disable horizontal scrollbar dragging for now unless we encounter a good reason for it.